Assets/MixedRealityToolkit/Services/MixedRealityToolkit.cs

```diff
@@ -1007,35 +1012,36 @@ private void DestroyAllServices()
 
         private bool ExecuteOnAllServicesInOrder(Action<IMixedRealityService> execute)
         {
-            if (!HasProfileAndIsInitialized)
-            {
-                return false;
-            }
-
-            var services = MixedRealityServiceRegistry.GetAllServices();
-            int length = services.Count;
-            for (int i = 0; i < length; i++)
-            {
-                execute(services[i]);
-            }
-
-            return true;
+            return ExecuteOnServiceCollection(
+                execute,
+                MixedRealityServiceRegistry.GetAllServices());
         }
 
         private bool ExecuteOnAllServicesReverseOrder(Action<IMixedRealityService> execute)
+        {
+            return ExecuteOnServiceCollection(
+                execute,
+                MixedRealityServiceRegistry.GetAllServices().Reverse());
+        }
+
+        private bool ExecuteOnServiceCollection(
+            Action<IMixedRealityService> execute,
+            IEnumerable<IMixedRealityService> services)
         {
             if (!HasProfileAndIsInitialized)
             {
                 return false;
             }
 
-            var services = MixedRealityServiceRegistry.GetAllServices();
-            int length = services.Count;
-
-            for (int i = length - 1; i >= 0; i--)
+            try
             {
-                execute(services[i]);
+                foreach (IMixedRealityService service in services)
+                {
+                    execute(service);
+                }
             }
+            catch (InvalidOperationException)
+            { /* The service collection changed */ }
```

Comment:
Please include the comments you had above regarding why you do this, since this is in another file and people won't know why it happens.

Reply:
Good catch. I intended to add that comment everywhere and missed this spot.